gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1036,6 +1026,101 @@ double ofx_get_investment_amount(const struct OfxTransactionData* data)
     }
 }
 
+// Forward declaration, required because several static functions depend on one-another.
+static void
+gnc_file_ofx_import_int (ofx_info* info);
+
+// gnc_ofx_process_next_file processes the next file in the info->file_iter list.
+static void
+gnc_ofx_process_next_file (GtkDialog *dialog, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    if (info->file_iter->next)
+    {
+        // Done with the previous OFX file, process the next one if any.
+        info->file_iter = info->file_iter->next;
+        gnc_file_ofx_import_int (info);
+    }
+    else
+    {
+        // No more OFX file to process, final cleanup.
+        g_slist_free_full (info->file_list, g_free);
+        g_free (info);
+        if (ofx_created_commodites)
+        {
+            /* FIXME: Present some result window about the newly created
+             * commodities */
+            g_warning ("Created %d new commodities during import", g_list_length (ofx_created_commodites));
+            g_list_free (ofx_created_commodites);
+            ofx_created_commodites = NULL;
+        }
+        else
+        {
+            //g_warning("No new commodities created");
+        }
+    }
+}
+
+
+// When the user is done matching transactions, trigger processing the next OFX file.
+static void
+gnc_ofx_match_done (GtkDialog *dialog, gint response_id, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    gnc_ofx_process_next_file (dialog, info);
+}
+
+// Aux function to process the OFX file in info->file_iter
+static void
+gnc_file_ofx_import_int (ofx_info* info)
+{
+    LibofxContextPtr libofx_context = libofx_get_new_context();
```

Comment:
Do you need to create a new context for each file?
If yes, I see no libofx_free_context
Maybe you can use the unused one in gnc_file_ofx_import and move the callback setup to there also ?

Reply:
There was no libofx_free_context() in the original code, I don't think (unless somehow I removed it)? I'll try to reuse the same one, that's cleaner.

Reply:
Possibly it relied on there being one matcher and it all got freed when the dialog closed or it was just missed and it should be there so may be worth trying to add it just for clarity.